File: leetcode/234e.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        nodes = []
        new_head = head
        while new_head:
            nodes.append(new_head)
            new_head = new_head.next
        
        new_head = head
        while new_head:
            the_other_head = nodes.pop()
            if new_head.val != the_other_head.val:
                return False
            new_head = new_head.next
        return True

File: leetcode/234e.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> lst = new ArrayList<>();
        ListNode cur = head;

        while (cur != null) {
            lst.add(cur.val);
            cur = cur.next;
        }

        // List<Integer> lst2 = lst.subList(0, lst.size());
        List<Integer> lst2 = new ArrayList(lst);
        Collections.reverse(lst2);
        // System.out.println(lst2);
        return lst.equals(lst2);
    }
}
